omc3/scripts/luminosity_imbalance.py

```diff
@@ -0,0 +1,175 @@
+
+        # Compute the relative error for this IP
+        rel_error[ip] = 0.5 * \
+            sqrt(sum([(error[p] ** 2) / (average_beta[p] ** 2) for p in PLANES]))
+
+        # Get the effective beta
+        eff_beta[ip] = sqrt(reduce(mul, average_beta.values()))
+
+    # Compute the whole relative error
+    relative_error = sum(rel_error.values())
+
+    # And get the imbalance
+    imbalance = eff_beta[IPS[0]] / eff_beta[IPS[1]]
+
+    return {'imbalance': imbalance,
+            'relative_error': relative_error,
+            'eff_beta_ip1': eff_beta[IPS[0]],
+            'rel_error_ip1': rel_error[IPS[0]],
+            'eff_beta_ip5': eff_beta[IPS[1]],
+            'rel_error_ip5': rel_error[IPS[1]]
+            }
+
+
+def print_luminosity(res):
+    LOG.info(f'{"Luminosity Imbalance":22s}: {res["imbalance"]}')
+    LOG.info(f'{"Relative error":22s}: {res["relative_error"]}')
+    LOG.info(f'{"Effective beta IP1":22s}: {res["eff_beta_ip1"]}')
+    LOG.info(f'{"Rel error IP1":22s}: {res["rel_error_ip1"]}')
+    LOG.info(f'{"Effective beta IP5":22s}: {res["eff_beta_ip5"]}')
+    LOG.info(f'{"Rel error IP5":22s}: {res["rel_error_ip5"]}')
+
+
+def get_significant_digits(res):
+    res['imbalance'], res['relative_error'] = significant_digits(
+            res['imbalance'], res['relative_error'])
+    res['eff_beta_ip1'], res['rel_error_ip1'] = significant_digits(
+            res['eff_beta_ip1'], res['rel_error_ip1'])
+    res['eff_beta_ip5'], res['rel_error_ip5'] = significant_digits(
+            res['eff_beta_ip5'], res['rel_error_ip5'])
+    return res
+
+
+def merge_tfs(directories, filename):
+    # Combine the data into one tfs
+    new_tfs = tfs.TfsDataFrame()
+    for d in sorted(directories):
+        path = d / filename
+        data = tfs.read_tfs(path)
+
+        new_tfs = new_tfs.append(data, ignore_index=True)
+
+    return new_tfs
+
+
+def get_ip_dir_names(kmod_dirs):
+    pattern = re.compile(".*ip[0-9]B[1-2]")
+    ip_dir_names = [d for kmod in kmod_dirs
+                    for d in kmod.glob('**/*')
+                    if pattern.match(d.name) and d.is_dir()]
+
+    if len(ip_dir_names) != 4:
+        raise Exception('All directories should account for a total of 4 ipBx '
+                        'directories inside')
+
+    return ip_dir_names
+
+
+def loader_params():
+    params = EntryPointParameters()
+    params.add_parameter(name="kmod_dirs", type=pathlib.Path,
+                         nargs='+', required=True,
+                         help="Path to kmod directories with stored KMOD "
+                              "measurement files")
+    params.add_parameter(name="res_dir", type=pathlib.Path, required=True,
+                         help="Output directory where to write the result tfs")
+    return params
+
+
+@entrypoint(loader_params(), strict=True)
+def merge_and_copy_kmod_output(opt):
+    # Get the directories we need where the tfs are stored
+    ip_dir_names = get_ip_dir_names(opt.kmod_dirs)
```

Comment:
I'd have another input argument to set the data-labels manually (so you don't have to rename the folders) and only if that is not given to call this function